docs/running-on-mesos.md

```diff
@@ -702,7 +702,16 @@ See the [configuration page](configuration.html) for information on Spark config
     Set the maximum number GPU resources to acquire for this job. Note that executors will still launch when no GPU resources are found
     since this configuration is just an upper limit and not a guaranteed amount.
   </td>
-  </tr>
+</tr>
+<tr>
+  <td><code>spark.mesos.disk</code></td>
+  <td><code>0</code></td>
```

Comment:
If it's at the default of 0, no behavior changes right? i.e. it's not required to set this in general, only in the case you cite?

Reply:
Well, as you can see in https://github.com/criteo-forks/mesos/blob/3de5efba936c8b7bd1bf88c2fd05006a93271b73/src/common/http.cpp#L725,  the Mesos API returns a default value of 0 if no disk is provided.

So as far as I'm concerned it should be ok but since you asked let me do the fix to avoid providing any disk in the TaskInfo if not specified in the conf. That way we'll be sure that Spark remains compatible if the behavior changes on Mesos side (i.e., if it becomes "no value is not equivalent to 0").

Reply:
I actually protected the code by inserting the disk amount only if it is > 0. But anyway I think it's better to distinguish between None and 0 because then the Mesos community can decide to change their mind at and this won't have any impact on Spark.
Fix is coming in a minute.